### PR TITLE
MCH: various fixes to MCH workflows

### PR DESCRIPTION
**Pull request by aferrero2707**
- removed references to specific RDH versions
- removed obsolete file-to-digits workflow
- added cru-page-to-digits workflow, taking the output of the cru-page-reader workflow
- input spec of raw-to-digits workflow set to "TF:MCH/RAWDATA"

### CODE REVIEW

File: Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx [hunk 1]
```diff
@@ -251,7 +246,7 @@ class DataDecoderTask
 
     for (auto it = parser.begin(), end = parser.end(); it != end; ++it) {
       // retrieving RDH v4
```
Comment (aphecetche) [inline Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx]: The v4 can go away in the comment. In addition I'd say the full comment can go away (the one below about size as well), as the code is descriptive enough without the comment ?

Comment (aferrero2707) [reply Detectors/MUON/MCH/Workflow/src/DataDecoderSpec.cxx]: I agree, the comments are now removed.

**Review by aphecetche — APPROVED** [review]

**Comment by aferrero2707** [comment]
@aphecetche @ktf once again, the automatic checks are failing due to compiler crashes and not code errors. In this specific case:
```
[  3%] Building CXX object Utilities/rANS/CMakeFiles/O2test-rans-EncodeDecode.dir/test/test_ransEncodeDecode.cxx.o
[  4%] Building CXX object Utilities/DataCompression/CMakeFiles/O2test-datacompression-DataDeflater.dir/test/test_DataDeflater.cxx.o
c++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <https://gcc.gnu.org/bugs/> for instructions.
gmake[2]: *** [DataFormats/MemoryResources/CMakeFiles/O2test-memoryresources-MemoryResources.dir/test/testMemoryResources.cxx.o] Error 4
gmake[1]: *** [DataFormats/MemoryResources/CMakeFiles/O2test-memoryresources-MemoryResources.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
```

Is there a way to re-start the check jobs without inserting a dummy commit?

**Comment by chiarazampolli** [comment]
I see the same error in my PR (https://github.com/AliceO2Group/AliceO2/pull/4471), plus another one that you do not have in the clang-format checks...

**Comment by aferrero2707** [comment]
The macOS build fails due to one broken check:
```
=== o2sim_hepmc - /Environment ===
=== o2sim_hepmc - Will not get stack trace of processes timing out: pstack not found ===
=== o2sim_hepmc - Timeout prefix: timeout --signal=SIGTERM --kill-after=10s 400s ===
=== o2sim_hepmc - Running /System/Volumes/Data/build/ci_checks/o2_checker_macos_o2_ci_0/sw/BUILD/bceeededdd5ac1a5d1059b98a096c8a63b3ccbb2/O2/stage/bin/o2-sim with args -n 2 -j 2 -g hepmc --configKeyValues HepMC.fileName=/System/Volumes/Data/build/ci_checks/o2_checker_macos_o2_ci_0/sw/SOURCES/O2/4493/0/Generators/share/data/pythia.hepmc;HepMC.version=2 -o o2simhepmc (attempt 1/3) ===
[INFO] Running with 2 sim workers 
[INFO] CREATING SIM SHARED MEM SEGMENT FOR 2 WORKERS
shmget: shmget failed: Invalid argument
[INFO] SHARED MEM INITIALIZED AT ID -1
[WARN] COULD NOT CREATE SHARED MEMORY ... FALLING BACK TO SIMPLE MODE
Spawning particle server on PID 69465; Redirect output to serverlog
Spawning sim worker 0 on PID 69466; Redirect output to workerlog0
Spawning hit merger on PID 69467; Redirect output to mergerlog
[INFO] DISTRIBUTING EVENT : 1
[INFO] DISTRIBUTING EVENT : 2
[INFO] EVENT FINISHED : 1
[INFO] Merger process 69467 returned
[INFO] Simulation process took 45.8827 s
Error in <TSystem::ExpandFileName>: input: $ROOT_DYN_PATH/libRIO, output: $ROOT_DYN_PATH/libRIO
Error in <TSystem::ExpandFileName>: input: $ROOT_DYN_PATH/libCling, output: $ROOT_DYN_PATH/libCling
Warning in <TFile::Init>: file o2simhepmc_Kine.root probably not closed, trying to recover
Warning in <TFile::Init>: no keys recovered, file has been made a Zombie
=== o2sim_hepmc - Test attempt 1/3 failed with exit code 1 ===
```

**Pull request merged**